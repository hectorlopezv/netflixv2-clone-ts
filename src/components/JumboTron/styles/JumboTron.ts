import styled from 'styled-components/macro';

interface typesProps {
    direction : string;
}

export const Inner = styled.div<typesProps>`
    display: flex;
    flex-direction: ${ ({direction}) => direction  };
    align-items: center;
    justify-content: space-between;
    max-width: 1100px;
    margin: auto;
    width: 100%;

    @media (max-width: 1000px){
        flex-direction: column;
    }
`;


export const Container =  styled.div<>``;

export const JumboTitle = styled.h1<>``;



export const JumboSubTitle = styled.div<>``;


interface imageProps {
    src: string;
    alt: string;
}
export const JumboImage = styled.div<imageProps>``;