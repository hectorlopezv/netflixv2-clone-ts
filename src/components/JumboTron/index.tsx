import React from 'react';

//imports styles components
import { Inner, Container, Image, Title, SubTitle, Pane, Item} from './styles/JumboTron';

export interface JumboTronProps {
    direction: string;
}
 
const JumboTron: React.FC<JumboTronProps> = ({direction = 'row', children, ...props}) => {
    return ( 
            <Item >
                <Inner direction={direction}>
                     {children}
                </Inner>
            </Item>
     );
}
 
export default JumboTron;



export interface JumboContainerProps {
}


export const Jumbo_Container: React.FC<JumboContainerProps> = ({children, ...props}) => {
    return (  
        <Container {...props}>{children}</Container>
    );
}

export interface JumbOTitleProps {
    
}
 

export const Jumbo_Title: React.FC<JumbOTitleProps> = ({children, ...props}) => {
    return (  
        <Title {...props}>{children}</Title>
      
    );
}
 
export interface JumboSubTitleProps {
    
}
 
export const Jumbo_SubTitle: React.FC<JumboSubTitleProps> = ({children, ...props}) => {
    return ( 

        <SubTitle {...props}>{children}</SubTitle>
    
     );
}

export interface JumboImageProps {
    src: string;
    alt: string;
    
}
 
export const Jumbo_Image: React.FC<JumboImageProps> = ({...props}) => {
    return (

        <Image {...props}/>
    
      );
}
 
export interface JumboTron_PaneProps {
    
}
 
export const JumboTron_Pane: React.FC<JumboTron_PaneProps> = ({children, ...props}) => {
    return (  

        <Pane {...props}>{children}</Pane>
    );
}
 

 
