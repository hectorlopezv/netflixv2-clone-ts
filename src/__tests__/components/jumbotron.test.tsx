/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import { render, fireEvent, getByTestId, getByAltText } from '@testing-library/react';

import JumboTron, { Jumbo_Container, Jumbo_Title, Jumbo_SubTitle, Jumbo_Image, JumboTron_Pane } from '../../components/JumboTron';
import dataJumbo from '../../fixtures/jumbo.json';

describe('JumboTron', () => {
    it('JumboTron <JumboTron /> render with populated data', () => {
        const {container, getByText, getByAltText, getByTestId} = render(
            <Jumbo_Container>
                {dataJumbo.map((j: any) => (
                <JumboTron key={j.id} direction={j.direction}>
                    <JumboTron_Pane>
                    <Jumbo_Title>{j.title}</Jumbo_Title>
                    <Jumbo_SubTitle>{j.subTitle}</Jumbo_SubTitle>
                    </JumboTron_Pane>
        
                    <JumboTron_Pane>
                    <Jumbo_Image src={j.image} alt={j.alt} data-testid={`${j.id}-jumbo-image`} />
                    </JumboTron_Pane>
        
                </JumboTron>
                ))}
            </Jumbo_Container>
        );

        //assertions
        expect(getByText('Enjoy on your TV.')).toBeTruthy();
        expect(getByAltText('Tiger King on Netflix')).toBeTruthy();
        expect(getByTestId('1-jumbo-image')).toBeTruthy();
        expect(getByText('Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.')).toBeTruthy();
        
        expect(container.firstChild).toMatchSnapshot();
        

    });
});