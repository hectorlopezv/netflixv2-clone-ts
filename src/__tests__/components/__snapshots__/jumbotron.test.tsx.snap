// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JumboTron JumboTron <JumboTron /> render with populated data 1`] = `
<div
  class="JumboTron__Container-sc-1k9j6n2-6 fFMXoq"
>
  <div
    class="JumboTron__Item-sc-1k9j6n2-0 bOZjRh"
  >
    <div
      class="JumboTron__Inner-sc-1k9j6n2-1 cRuNOx"
      direction="row"
    >
      <div
        class="JumboTron__Pane-sc-1k9j6n2-5 booXfB"
      >
        <h1
          class="JumboTron__Title-sc-1k9j6n2-2 gaotQX"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="JumboTron__SubTitle-sc-1k9j6n2-3 duPSiL"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="JumboTron__Pane-sc-1k9j6n2-5 booXfB"
      >
        <img
          alt="Tiger King on Netflix"
          class="JumboTron__Image-sc-1k9j6n2-4 guSpEf"
          data-testid="1-jumbo-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="JumboTron__Item-sc-1k9j6n2-0 bOZjRh"
  >
    <div
      class="JumboTron__Inner-sc-1k9j6n2-1 kOYazw"
      direction="row-reverse"
    >
      <div
        class="JumboTron__Pane-sc-1k9j6n2-5 booXfB"
      >
        <h1
          class="JumboTron__Title-sc-1k9j6n2-2 gaotQX"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="JumboTron__SubTitle-sc-1k9j6n2-3 duPSiL"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="JumboTron__Pane-sc-1k9j6n2-5 booXfB"
      >
        <img
          alt="Watch on mobile"
          class="JumboTron__Image-sc-1k9j6n2-4 guSpEf"
          data-testid="2-jumbo-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="JumboTron__Item-sc-1k9j6n2-0 bOZjRh"
  >
    <div
      class="JumboTron__Inner-sc-1k9j6n2-1 cRuNOx"
      direction="row"
    >
      <div
        class="JumboTron__Pane-sc-1k9j6n2-5 booXfB"
      >
        <h1
          class="JumboTron__Title-sc-1k9j6n2-2 gaotQX"
        >
          Watch everywhere.
        </h1>
        <h2
          class="JumboTron__SubTitle-sc-1k9j6n2-3 duPSiL"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="JumboTron__Pane-sc-1k9j6n2-5 booXfB"
      >
        <img
          alt="Money Heist on Netflix"
          class="JumboTron__Image-sc-1k9j6n2-4 guSpEf"
          data-testid="3-jumbo-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
